enumerables.rb

```diff
@@ -0,0 +1,102 @@
+module Enumerable
+  def my_each
+    size = count - 1
+    counter = 0
+    items = to_a
+    while counter <= size
+      if is_a? Range
+        items.length >= 0 && yield(items[counter])
+      elsif is_a? Hash
+        yield(items[counter][0], items[counter][1])
+      else
+        yield(self[counter])
+      end
+      counter += 1
+    end
+    self
+  end
```

Comment:
- [ ] should be identical and returns the same thing as ruby's [`each`](https://ruby-doc.org/core-2.6.4/Array.html#method-i-each). ([Screenshot](https://gitlab.com/microverse/guides/projects/requirements_screenshots/raw/master/images/ruby/advanced_building_blocks_enumerable/my_each.png) from [Odin](https://www.theodinproject.com/courses/ruby-programming/lessons/advanced-building-blocks#assignment-2))
- [ ] my_each returns an Enumerator if no block is given
```ruby
array = Array.new(100) { rand(0...9) } }
array.my_each) == Enumerator #1 true
```

Reply:
Hi @whiz25, I don't understand what exactly you want me to do here, because i evn tested it with normal `each` method and got a `a false`
![image](https://user-images.githubusercontent.com/46686100/75692037-3d897800-5ca5-11ea-9b7c-5b075cdab35f.png)